pkg/provisioner/ironic/findhost_test.go

```diff
@@ -29,15 +29,15 @@ func TestFindExistingHost(t *testing.T) {
 			hostName:       "name",
 			provisioningID: "uuid",
 			ironic: testserver.New(t, "ironic").NotFound("/v1/nodes/uuid").
-				Response("/v1/nodes/name", `{"name": "name", "uuid": "different-uuid"}`),
+				Response("/v1/nodes/name", `{"name": "name", "uuid": "different-uuid"}`, 200),
```

Comment:
In my latest changes there's a `ResponseJSON` that could be useful for that. In addition, I'd suggest to enrich the IronicMock public interface with a "speaking" method (ie `ChangeNodeName`) and use `Response` just in the implementation